src/main/java/dinkplugin/notifiers/PetNotifier.java

```diff
@@ -215,56 +236,316 @@ private static class ParseResult {
+                @Override
+                Double getProbability(Client client, KillCountService kcService) {
+                    // https://oldschool.runescape.wiki/w/Ancient_chest#Unique_drop_table
+                    int totalPoints = client.getVarbitValue(Varbits.TOTAL_POINTS);
+                    if (totalPoints <= 0) {
+                        totalPoints = 26_025;
+                    }
+                    return 0.01 * (totalPoints / 8_676) / 53;
+                }
+
+                @Override
+                Integer estimateActions(Client client, KillCountService kcService) {
+                    return Stream.of("", " Challenge Mode")
+                        .map(suffix -> "Chambers of Xeric" + suffix)
+                        .map(event -> kcService.getKillCount(LootRecordType.EVENT, event))
+                        .filter(Objects::nonNull)
+                        .reduce(Integer::sum)
+                        .orElse(null);
+                }
+            }),
+            entry("Pet chaos elemental", new MultiKcSource("Chaos Elemental", 1.0 / 300, "Chaos Fanatic", 1.0 / 1_000)),
+            entry("Pet dagannoth prime", new KcSource("Dagannoth Prime", 1.0 / 5_000)),
+            entry("Pet dagannoth rex", new KcSource("Dagannoth Rex", 1.0 / 5_000)),
+            entry("Pet dagannoth supreme", new KcSource("Dagannoth Supreme", 1.0 / 5_000)),
+            entry("Pet dark core", new KcSource("Corporeal Beast", 1.0 / 5_000)),
+            entry("Pet general graardor", new KcSource("General Graardor", 1.0 / 5_000)),
+            entry("Pet k'ril tsutsaroth", new KcSource("K'ril Tsutsaroth", 1.0 / 5_000)),
+            entry("Pet kraken", new KcSource("Kraken", 1.0 / 5_000)),
+            entry("Pet penance queen", new Source() {
+                @Override
+                Double getProbability(Client client, KillCountService kcService) {
+                    return 1.0 / 1_000;
+                }
+
+                @Override
+                Integer estimateActions(Client client, KillCountService kcService) {
+                    return client.getVarbitValue(Varbits.BA_GC);
+                }
+            }),
+            entry("Pet smoke devil", new KcSource("Thermonuclear smoke devil", 1.0 / 3_000)),
+            entry("Pet snakeling", new KcSource("Zulrah", 1.0 / 4_000)),
+            entry("Pet zilyana", new KcSource("Commander Zilyana", 1.0 / 5_000)),
+            entry("Phoenix", new KcSource("Wintertodt", 1.0 / 5_000)),
+            entry("Prince black dragon", new KcSource("King Black Dragon", 1.0 / 3_000)),
+            entry("Quetzin", new Source() {
+                @Override
+                Double getProbability(Client client, KillCountService kcService) {
+                    return null; // currently unknown
+                }
+
+                @Override
+                Integer estimateActions(Client client, KillCountService kcService) {
+                    return null; // currently untracked
```

Comment:
this can now be tracked, but there's no point yet since the probabilities are unknown https://github.com/runelite/runelite/commit/43f67f4d89a9744cd26ca017b2cd7e6c32d35ed7

Reply:
Shall we just infer the probability from the [official stats](https://cdn.runescape.com/assets/img/external/oldschool/2024/newsposts/2024-03-26/Varlamore_Stats.png) (and assume same rate across expert/master tiers)? & of course can be adjusted when more concrete information is available

Reply:
Done in https://github.com/pajlads/DinkPlugin/pull/433/commits/9d60269cad2a3f120f5622714814926e7353dd51